// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

//option java_multiple_files = true;
//option java_package = "io.grpc.examples.helloworld";
//option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package Database;

// The query-and-insert service definition.
service QueryHandler {
  rpc Query (QueryRequest) returns (QueryReply) {}

  rpc QueryInsert (QueryRequest) returns (QueryReply) {}

  rpc QueryInsertTimeStamp (QueryRequestTimeStamp) returns (QueryReply) {}

  rpc QueryInsertFrameID (QueryRequestFrameID) returns (QueryReply) {}
}

// The request message containing the user's name.
message QueryRequest {
  int32 frame_id = 1;
  int32 cls_id = 2;
  int32 num_query = 3;
  repeated float query = 4;
  repeated float center = 5;
  // Indexes that has been taken by tracked ones
  // will be single -1 if no void index exists.
  repeated int64 matched_indexes = 6;
}

// Time stamp message
message TimeStamp {
  int64 sec = 1;
  int64 usec = 2;
}

// The request message containing the user's name.
message QueryRequestTimeStamp {
  int32 server_id = 1;
  TimeStamp ts = 2;
  QueryRequest qr = 3;
}

// The request message containing the user's name.
message QueryRequestFrameID {
  int32 server_id = 1;
  int32 frame_id = 2;
  QueryRequest qr = 3;
}

// The response message containing the greetings
message QueryReply {
  string status = 1;
  repeated int64 indexes = 2;
}

